birb_registry/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+#![feature(const_trait_impl)]
+#![feature(fs_try_exists)]
+
+use birb::MainThreadModule;
+use std::{collections::HashMap, io::prelude::*, fs::File, fs};
+use std::fmt::{Debug, Formatter};
+use serde::{Serialize, Deserialize};
+
+#[derive(Default)]
+pub struct Registry {
+    ticks: u16,
+    data: HashMap<String, Vec<u8>>
+}
+
+// Implement Registry Methods
+impl Registry {
+    pub fn new() -> Self {
+        Registry::default()
+    }
+    pub fn store<T>(&mut self, key: String, object: &T) where T: Serialize {
+        self.data.insert(key, serde_json::to_string(object).unwrap().into_bytes());
+    }
+    pub fn get<T>(&self, key: String) -> T where T: for<'a> Deserialize<'a> {
+        let object: T = serde_json::from_slice(self.data.get(&key).unwrap()).unwrap();
+        return object
+    }
+}
+
+impl Debug for Registry {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Registry")
+            .field("data", &self.data)
+            .finish()
+    }
+}
+
+impl MainThreadModule for Registry {
+    fn tick(&mut self, _: &birb::MainThreadApp) {
+        // shitty persistence :3
+        // loads json on zero
+        if self.ticks == 0 {
```

Comment:
This should be put in a load function rather than being based on some timing system